Sources/IMADAIPlugin.swift

```diff
@@ -404,33 +407,28 @@ import PlayKitUtils
         
         invalidateRequestTimer()
         streamManager = adsLoadedData.streamManager
-        adsLoadedData.streamManager.delegate = self
+        adsLoadedData.streamManager?.delegate = self
         
         createRenderingSettings()
         
         streamManager?.initialize(with: renderingSettings)
     }
     
-    public func adsLoader(_ loader: IMAAdsLoader!, failedWith adErrorData: IMAAdLoadingErrorData!) {
+    public func adsLoader(_ loader: IMAAdsLoader, failedWith adErrorData: IMAAdLoadingErrorData) {
         // Cancel the request timer
         invalidateRequestTimer()
         stateMachine.set(state: .adsRequestFailed)
         
-        guard let adError = adErrorData.adError else {
-            PKLog.error("AdsLoader faild with error.")
-            return
-        }
-        let adErrorMessage: String = adError.message == nil ? "" : adError.message
-        PKLog.error(adErrorMessage)
-        messageBus?.post(AdEvent.Error(nsError: IMAPluginError(adError: adError).asNSError))
-        delegate?.adsPlugin(self, loaderFailedWith: adErrorMessage)
+        PKLog.error(adErrorData.adError.message)
```

Comment:
@chausov that mathod may have error or not why did you remove the validation if there is error or not?

Reply:
adErrorData now is non Optional type so `guard let adError = adErrorData.adError` does not make a sense